sycl/doc/extensions/proposed/SYCL_EXT_ONEAPI_DEVICE_GLOBAL.asciidoc

```diff
@@ -706,6 +706,34 @@ of type `device_global` in which case it can be odr-used inside a device functio
 global variable that isn't `const` or `constexpr` unless the variable is of type `device_global`.
 
 
+=== Referencing a device global defined in another translation unit
+
+If the implementation defines the `SYCL_EXTERNAL` macro, device code in one
+translation unit may reference a device global variable that is defined in a
+different translation unit so long as the variable is defined using
+`SYCL_EXTERNAL`.  For example:
+
+```c++
+// In one translation unit
+#include <sycl/sycl.hpp>
+using namespace sycl::ext::oneapi;
+
+SYCL_EXTERNAL device_global<int> Foo;  // definition
+
+// In another translation unit
+#include <sycl/sycl.hpp>
+using namespace sycl::ext::oneapi;
+using namespace sycl;
+
+extern device_global<int> Foo;  // declaration
```

Comment:
I think we have a diagnostic that forces users to add SYCL_EXTERNAL macro to undefined functions that are called from device context. Should we add this requirement for global variables as well?

Reply:
Thanks for pointing this out.  This actually contradicts the SYCL 2020 spec, which seems to say that `SYCL_EXTERNAL` is required only on the definition of the function.  However, I like the DPC++ behavior better, and we have an open Khronos issue asking to clarify the wording about `SYCL_EXTERNAL` in this regard.  I just added a note to that issue indicating that it would be better to require `SYCL_EXTERNAL` on both the function definition and the declaration (i.e. matching the DPC++ behavior).

So, I agree that we should change the wording here in the device global spec too.  I did this in 379b314d23e94efd2c6356830397f21f77720caa.